src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -829,108 +830,132 @@ subroutine init_atm_static(mesh, dims, configs)
-             lon_pt = start_lon + (supersample_fac*(iTileStart-1) + i - (supersample_fac*3+1)) * 0.05 / supersample_fac
-             lat_pt = lat_pt * PI / 180.0
-             lon_pt = lon_pt * PI / 180.0
+            if (tile % is_processed) then
+                cycle
+            end if
 
-             iPoint = nearest_cell(lat_pt,lon_pt,iPoint,nCells,maxEdges, &
-                                   nEdgesOnCell,cellsOnCell, &
-                                   latCell,lonCell)
-             if (rarray(ii,jj,1) /= msgval) then
+            call mpas_log_write('Processing tile: '//trim(tile % fname))
+
+            all_pixels_mapped_to_halo_cells = .true.
+
+            do j = supersample_fac * tile_bdr + 1, supersample_fac * (tile_ny + tile_bdr), 1
+                do i = supersample_fac * tile_bdr + 1, supersample_fac * (tile_nx + tile_bdr), 1
+
+                    ii = (i - 1) / supersample_fac + 1
+                    jj = (j - 1) / supersample_fac + 1
+
+                    call mgr % tile_to_latlon(tile, j, i, lat_pt, lon_pt, supersample_fac)
+                    call mpas_latlon_to_xyz(xPixel, yPixel, zPixel, sphere_radius, lat_pt, lon_pt)
+                    call mpas_kd_search(tree, (/xPixel, yPixel, zPixel/), res)
+
+                    if (tile % tile(ii, jj, 1) * scalefactor /= missing_value) then
+                        !
+                        ! This field only matters for land cells, and for all but the outermost boundary cells,
+                        ! we can safely assume that the nearest model grid cell contains the pixel (else, a different
+                        ! cell would be nearest)
+                        !
+                        if (landmask(res % id) == 1 .and. bdyMaskCell(res % id) < nBdyLayers) then
+                            snoalb(res % id) = snoalb(res % id) + tile % tile(ii, jj, 1)
+                            nhs(res % id) = nhs(res % id) + 1
+                            if (res % id <= nCellsSolve) then
+                                all_pixels_mapped_to_halo_cells = .false.
+                            end if
+                        ! For outermost land cells, additional work is needed to verify that the pixel
+                        ! actually lies within the nearest cell
+                        else if (landmask(res % id) == 1) then
+                             if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                              nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+
+                                    snoalb(res % id) = snoalb(res % id) + tile % tile(ii, jj, 1)
+                                    nhs(res % id) = nhs(res % id) + 1
+
+                                if (res % id <= nCellsSolve) then
+                                    all_pixels_mapped_to_halo_cells = .false.
```

Comment:
I think the logic to set `all_pixels_mapped_to_halo_cells` to `.false.` can only happen if `landmask(res % id) == 1`. Would this cause erroneous results if some pixel from a tile did map to a non-halo cell, but that non-halo cell was a water cell?

Reply:
I think code at lines [915 and 916](https://github.com/MiCurry/MPAS-Model/blob/92bbe60116ab64d7cfe06aec51304f80fdb6d203/src/core_init_atmosphere/mpas_init_atm_static.F#L915-L917) can only occur when landmask(res % id) == 1, because of the `else if` statement at line [908](https://github.com/MiCurry/MPAS-Model/blob/92bbe60116ab64d7cfe06aec51304f80fdb6d203/src/core_init_atmosphere/mpas_init_atm_static.F#L908). Does that seem correct to you? Or, perhaps I am missing something?

Reply:
Let me try to ask the question a different way: what if we did have pixels from a tile that mapped only to non-halo _water_ cells? Would `all_pixels_mapped_to_halo_cells` be correctly set in that case?

Reply:
Ah, yes I see you what you mean and you are right. I'll rearrange the logic to make sure `all_pixels_mapped_to_halo_cells` is correctly set.